provider/core/src/datagen/mod.rs

```diff
@@ -57,7 +57,7 @@ pub trait DataExporter: Sync {
         payload: &DataPayload<ExportMarker>,
     ) -> Result<(), DataError> {
         self.put_payload(key, &Default::default(), payload)?;
-        self.flush_with_fallback(key, FallbackMode::None)
+        self.flush_with_fallback(key, FallbackMode::External)
     }
 
```

Comment:
Discussion: `flush` and `flush_with_built_in_fallback`. Default impl for `flush` is the same, but for `flush_with_...`, return DataError.